### PR TITLE
fix: the DataFile version doesn't respect the writer's version

### PR DESCRIPTION
**Pull request by BubbleCal**

### CODE REVIEW

File: rust/lance-file/src/v2/writer.rs [hunk 1]
```diff
@@ -137,6 +137,13 @@ impl FileWriter {
         }
     }
 
+    /// Returns the format version that will be used when writing the file
+    pub fn version(&self) -> LanceFileVersion {
+        self.options
+            .format_version
+            .unwrap_or(LanceFileVersion::Stable)
```
Comment (westonpace) [inline rust/lance-file/src/v2/writer.rs:144]: For consistency we should probably do `.unwrap_or(LanceFileVersion::default_v2())` which is what we're using in `initialize`.  Although it's pretty much the same thing.

**Review by westonpace — APPROVED** [review]
Was it not writing anything `(0, 0)` or was it writing `(0, 3)`?  Because `0.3` and `2.0` are  aliases of each other.

For example:

```
    pub fn try_from_major_minor(major: u32, minor: u32) -> Result<Self> {
        match (major, minor) {
            (0, 0) => Ok(Self::Legacy),
            (0, 1) => Ok(Self::Legacy),
            (0, 2) => Ok(Self::Legacy),
            (0, 3) => Ok(Self::V2_0),
            (2, 0) => Ok(Self::V2_0),
            (2, 1) => Ok(Self::V2_1),
            _ => Err(Error::InvalidInput {
                source: format!("Unknown Lance storage version: {}.{}", major, minor).into(),
                location: location!(),
            }),
        }
    }
```

Either way, I'm ok with this change.

**Comment by westonpace** [comment]
FYI, it looks like this broke one of the python tests

**Comment by BubbleCal** [comment]
> Was it not writing anything `(0, 0)` or was it writing `(0, 3)`? Because `0.3` and `2.0` are aliases of each other.
> 
> For example:
> 
> ```
>     pub fn try_from_major_minor(major: u32, minor: u32) -> Result<Self> {
>         match (major, minor) {
>             (0, 0) => Ok(Self::Legacy),
>             (0, 1) => Ok(Self::Legacy),
>             (0, 2) => Ok(Self::Legacy),
>             (0, 3) => Ok(Self::V2_0),
>             (2, 0) => Ok(Self::V2_0),
>             (2, 1) => Ok(Self::V2_1),
>             _ => Err(Error::InvalidInput {
>                 source: format!("Unknown Lance storage version: {}.{}", major, minor).into(),
>                 location: location!(),
>             }),
>         }
>     }
> ```
> 
> Either way, I'm ok with this change.

it wrote (0,3) for v2 and (0,0) for v1.

File: rust/lance-file/src/v2/writer.rs (hunk 1 above)
Comment (BubbleCal) [reply rust/lance-file/src/v2/writer.rs]: Fixed

**Pull request merged**